### PR TITLE
std: Destabilize the internals of panic!

### PR DESCRIPTION
**Pull request by alexcrichton**
Now that we have a `#[allow_internal_unstable]` attribute for macros there's no
need for these two `begin_unwind` functions to be stable. Right now the `panic!`
interface is the only one we wish to stabilize, so remove the stability markers
from these functions.

While this is a breaking change, it is highly unlikely to break any actual code.
It is recommended to use the `panic!` macro instead if it breaks explicit calls
into `std::rt`.

[breaking-change]
cc #24208

### CODE REVIEW

File: src/libstd/rt/unwind.rs [hunk 1]
```diff
@@ -495,7 +495,6 @@ pub extern fn rust_begin_unwind(msg: fmt::Arguments,
 /// on (e.g.) the inlining of other functions as possible), by moving
 /// the actual formatting into this shared place.
 #[inline(never)] #[cold]
-#[stable(since = "1.0.0", feature = "rust1")]
```
Comment (sfackler) [inline src/libstd/rt/unwind.rs]: Do these need explicit `#[unstable]` annotations?

Comment (huonw) [inline src/libstd/rt/unwind.rs]: `unstable` inherits (in this case from `std::rt`).

**Comment by rprichard** [comment]
Aren't the pretty printing tests going to fail?   #23616.

**Comment by alexcrichton** [comment]
@rprichard you are most correct!

re-r? @huonw, I just deleted `// pretty-expanded` from all the relevant tests.

**Pull request merged**